native_client/dotnet/DeepSpeechClient/DeepSpeech.cs

```diff
@@ -52,14 +55,17 @@ public DeepSpeech()
                             aBeamWidth,
                             ref _modelStatePP);
             EvaluateResultCode(resultCode);
+            _activeModel = true;
         }
 
         /// <summary>
         /// Return the sample rate expected by the model.
         /// </summary>
         /// <returns>Sample rate.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when the acoustic model hasn't been initialized <see cref="CreateModel(string, uint)"/>.</exception>
         public unsafe int GetModelSampleRate()
         {
+            CheckAcousticModel();
```

Comment:
Can someone really call this `GetModelSampleRate` without having a `DeepSpeechClient::DeepSpeech` instance ? If that's possible, it does look very weird :)

Reply:
By just creating an instance of `DeepSpeechClient::DeepSpeech` yes, only creating the instance does not initialize the model, to initialize the model needs to call `CreateModel`, following dependency pattern the constructor should be simple and fast and any other data should pass throw object methods.

If you want to move the model creation to the construcor I'm ok with that, by moving it we prevent calling `GetModelSampleRate ` for instance without a model.

Reply:
> By just creating an instance of `DeepSpeechClient::DeepSpeech` yes, only creating the instance does not initialize the model, to initialize the model needs to call `CreateModel`, following dependency pattern the constructor should be simple and fast and any other data should pass throw object methods.

Ok, I missed that. Is this explicitely mentionned in the doc ? That's a slight variation of behavior regarding the other bindings, so it's a bit weird.



> If you want to move the model creation to the construcor I'm ok with that, by moving it we prevent calling `GetModelSampleRate ` for instance without a model.

Well, you are managing the code, and you have the .Net experience :). I'm just trying to make sure we don't add un-necessary checks that may blowup upon refactoring that someone else than you may be doing, in the future.